pkg/controller/networkpolicy_reconciler.go

```diff
@@ -236,43 +238,54 @@ func (reconciler *NetworkpolicyReconciler) syncNetworkPolicyHandler(key string)
 	}
 	namespace, err := reconciler.kubeclientset.CoreV1().Namespaces().Get(namespaceName, meta_v1.GetOptions{})
 	if err != nil {
-		reconciler.log.Errorf("namespace '%s'does not exist", namespaceName)
+		reconciler.log.Errorf("namespace '%s' does not exist", namespaceName)
 		return nil
 	}
 
 	setting := reconciler.getDefaultNetworkpolicySetting(namespace)
-	npName := setting.value
+	npNames := strings.Split(setting.value, defaultNetworkPoiliciesDelimiter)
 
+	for _, npName := range npNames {
+		if err := reconciler.syncNetworkPolicy(npName, namespaceName, name, setting); err != nil {
+			reconciler.log.Errorln("Could not sync network policy '%s'", npName, err)
+		}
+	}
+
+	if err := reconciler.deleteMultipleDefaultNetworkPolicies(key, setting); err != nil {
+		reconciler.log.Errorln("Failed to delete network policies: ", setting.value)
+		return err
+	}
+
+	return nil
+}
+
+func (reconciler *NetworkpolicyReconciler) syncNetworkPolicy(npName string, namespaceName string, name string, setting Setting) error {
 	if _, ok := reconciler.defaultNetworkPolicies[npName]; !ok {
 		if err := reconciler.updateBuffer(npName); err != nil {
-
 			reconciler.log.Warnf("Failed to get default network policy '%s'", npName)
 			return nil
 		}
 	}
 
-	if name == reconciler.defaultNetworkPolicies[npName].GetName() {
```

Comment:
Here we check if the current network policy in the buffer is a np which we should reconcile. Why did you delete this check?

Reply:
You are totally correct. I misunderstood the functioning of this part. I changed the function: As only this one work-item is processed, one does not need to loop over all possible default network policies and just change the one policy in question (if needed).

Had some trouble with the unit-tests though and had to re-adjust the setup for some of my tests.